FastSurferCNN/utils/metrics.py

```diff
@@ -103,175 +107,317 @@ def precision_recall(
 
 class DiceScore:
     """
-    Accumulate the component of the dice coefficient i.e. the union and intersection.
-
-    Attributes
-    ----------
-    op : callable
-        A callable to update accumulator. Method's signature is `(accumulator, output)`.
-        For example, to compute arithmetic mean value, `op = lambda a, x: a + x`.
-    output_transform : callable, optional
-        A callable that is used to transform the
-        :class:`~ignite.engine.Engine`'s `process_function`'s output into the
-        form expected by the metric. This can be useful if, for example, you have a multi-output model and
-        you want to compute the metric with respect to one of the outputs.
-    device : str of torch.device, optional
-        Device specification in case of distributed computation usage.
-        In most of the cases, it can be defined as "cuda:local_rank" or "cuda"
-        if already set `torch.cuda.set_device(local_rank)`. By default, if a distributed process group is
-        initialized and available, device is set to `cuda`.
```

Comment:
It seems to me that this metrics functionality of this class was previously implemented here and is now lost. Was this ever really used before. We should double-check. It might have been used for multi-gpu support or something.

It seems to me like FastSurfer still uses this (so at a minimum, it needs to be adapted there)
https://github.com/Deep-MI/FastSurfer/blob/462b00dc920bc5666da8c6df601afb0595737312/FastSurferCNN/utils/meters.py#L75

This device variable is being passed through the pipeline including `utils/train.py` but it only seems to be `self.device` which is initialized in init
https://github.com/Deep-MI/FastSurfer/blob/5bd139e20d8a017057550f435da6234396219206/FastSurferCNN/train.py#L79

In the end you will need to test the train script with a minimal dataset to make sure it does not get broken. I will find such a dataset for you, but if can also be generated with `generate_dataset.py`